File: AbstractFactory/python-source/factories/CavesFactory.py
from rooms.CavesHardRoom import CavesHardRoom
from rooms.CavesNormalRoom import CavesNormalRoom
from rooms.HardRoom import HardRoom
from rooms.NormalRoom import NormalRoom

from factories.RoomFactory import RoomFactory


class CavesFactory(RoomFactory):
    def create_normal_room(self) -> NormalRoom:
        return CavesNormalRoom()

    def create_hard_room(self) -> HardRoom:
        return CavesHardRoom()

File: JavaSource/src/main/java/ru/akhcheck/patterns/abstractfactory/factories/CavesFactory.java
package ru.akhcheck.patterns.abstractfactory.factories;

import ru.akhcheck.patterns.abstractfactory.rooms.CavesHardRoom;
import ru.akhcheck.patterns.abstractfactory.rooms.CavesNormalRoom;
import ru.akhcheck.patterns.abstractfactory.rooms.HardRoom;
import ru.akhcheck.patterns.abstractfactory.rooms.NormalRoom;

/** Фабрика комнат пещер. */
public class CavesFactory implements RoomFactory {
    @Override
    public NormalRoom createNormalRoom() {
        return new CavesNormalRoom();
    }

    @Override
    public HardRoom createHardRoom() {
        return new CavesHardRoom();
    }
}
